quickwit/quickwit-common/src/net.rs

```diff
@@ -330,7 +330,7 @@ fn is_private_ip(ip_addr: &IpAddr) -> bool {
 /// - Its labels (characters separated by `.`) are not empty.
 /// - Its labels are 63 or fewer characters.
 /// - Its labels do not start or end with '-' or '.'.
-fn is_valid_hostname(hostname: &str) -> bool {
+pub fn is_valid_hostname(hostname: &str) -> bool {
```

Comment:
I wonder if it makes more sense to chop off the part after `.` instead of allowing FQDNs here.

Reply:
In cloud environments, a common practice is to set the hostname to `<instance ID>.<az>.<region>.<azure|aws|...>.<company domain>.`, so it might be nice to actually allow FQDNs.

Reply:
That's a good point. I don't have strong feelings about that, but besides consistency with linux standalone nodes that only return hostname, I think long node ids might hunt us later when we will want to show them in different management UIs.

Reply:
👍 Than my only concern is the doc updates here.